plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java

```diff
@@ -391,8 +391,8 @@ protected String getTableSchemaName(ResultSet resultSet)
                 return Optional.of(doubleColumnMapping());
 
             case Types.DECIMAL:
-                int precision = typeHandle.getRequiredColumnSize();
-                int decimalDigits = typeHandle.getDecimalDigits().orElseThrow(() -> new IllegalStateException("decimal digits not present"));
+                int precision = typeHandle.getColumnSize().orElse(38);
```

Comment:
Ah, I missed that Phoenix has limited precision too. In that case forget my comments about DecimalModule.

You will need to create a new method - `decimalTests` cannot be re-used for testing DECIMAL (without precision and scale) because the `trinoCreateAsSelect` would fail on them.

But yes, the general idea looks correct.

Reply:
After playing for a bit I think we can dependency on DecimalModule, I did take that dependency and added code to allow overflow. 
Can you please give this a look now?

Reply:
@hashhar

Reply:
@vincentpoon Do you have some time to review this PR?

Reply:
ping! @hashhar @lhofhansl